pool/pool.go

```diff
@@ -424,11 +446,20 @@ func (p *pool) DeleteObject(ctx context.Context, params *client.DeleteObjectPara
 	if err != nil {
 		return err
 	}
-	err = cp.client.DeleteObject(ctx, params, options...)
+
+	res, err := cp.client.DeleteObject(ctx, params, options...)
+	if err == nil {
+		// reflect status failures in err
+		err = apistatus.ErrFromStatus(res.Status())
```

Comment:
As far as I see, the pattern of calling `ErrFromStatus` is a default behavior when we don't care about details of the error. 

```go
res, err := client.Call(ctxm params)
if err == nil {
    err = apistatus.ErrFromStatus(res.Status())
}
```

It will be much cleaner, if this error handling will be done in the client. Also, it does not match with go error handling pattern, which may lead to developer mistakes.
```go
value, err = doSomething()
if err != nil {
    // process error
}
// process success
```

Reply:
Error handling is encapsulated in `client.Client` implementation. `client.New` represents the most basic implementation.

Reply:
I do not agree that `client.New` should provide most basic implementation because it is high level multi protocol library.
However there is a client pool which is sort of a wrapper around the client which is used in multiple application such as HTTP / S3 Gateways, CDN edge-manager, so it can represent user-friendly version of the client.